From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Abhilash G <abhilashg@nvidia.com>
Date: Wed, 7 Jun 2023 08:19:33 +0000
Subject: [PATCH 082/240] NVIDIA: SAUCE: scsi: ufs: core: Add config flag to
 findout ufs query API exported or not

Add config flag to find out that UFS query APIs are exported
or not from the given kernel.

Change-Id: I9ceef118ad20c4264fad797b1a8176bc411822a2
Signed-off-by: Abhilash G <abhilashg@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2917006
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/drivers/ufs/core/Kconfig b/drivers/ufs/core/Kconfig
index 817208ee64ec..ec26b5b00b81 100644
--- a/drivers/ufs/core/Kconfig
+++ b/drivers/ufs/core/Kconfig
@@ -50,3 +50,11 @@ config SCSI_UFS_HWMON
 	  a hardware monitoring device will be created for the UFS device.
 
 	  If unsure, say N.
+
+config SCSI_UFSHCD_QUERY_EXPORT
+	bool "UFS HCD QUERY APIS exported"
+	def_bool y
+	depends on SCSI_UFSHCD
+	help
+	  Export the  UFS HCD query APIs for provisioning the UFS. This config will
+	  tell whether APIs are exported or not.
-- 
2.55.0

